### PR TITLE
OCPBUGS#6756: Adds conditional updates in web console and some reorg

### PR DESCRIPTION
**Pull request by opayne1**
Adds conditional updates in web console and some reorg

Version(s):
main and 4.14

Issue:
https://issues.redhat.com/browse/OCPBUGS-5756

Link to docs preview:
https://63315--docspreview.netlify.app/openshift-enterprise/latest/updating/updating_a_cluster/updating-cluster-web-console

QE review:
- [x] QE has approved this change.

Additional information:
This PR also includes some reorg to the assembly. It moves prerequisites around as well.

### CODE REVIEW

File: modules/before-updating-web-console.adoc [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+//
+// * updating/updating_a_cluster/updating-cluster-web-console.adoc
+
+:_content-type: CONCEPT
+[id="before-updating-web-console{context}"]
+= Before updating using the web console
+
+Before updating using the web console, you should:
+
+* Have access to the cluster as a user with `cluster-admin` privileges.
```
Comment (skopacz1) [inline modules/before-updating-web-console.adoc:11]: This can probably be removed since it's already a prereq in all the modules that need it (though I'd double check whether it's also a necessary prereq for the module about changing the update server)

File: modules/before-updating-web-console.adoc [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+* Have a recent etcd backup in case your update fails and you must restore your cluster to a previous state.
```
Comment (skopacz1) [inline modules/before-updating-web-console.adoc:12]: I would double check with an SME, but I think this prereq might be moot if there's nothing the user could do to trigger a new backup. If that's true, this is more like "hope that your most recent backup is recent enough to revert to if disaster recovery is necessary."

One idea might be to keep this here but reword it to something like:
"Check to see that your etcd backup is recent enough in case your update fails and you must restore your cluster to a previous state." Plus maybe a sentence about how etcd periodically creates a new backup on its own?

File: modules/before-updating-web-console.adoc [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+* You must replace {op-system-base}7 workers with {op-system-base}8 or {op-system} workers before updating to {product-title} {product-version}. Red Hat does not support in-place {op-system-base}7 to {op-system-base}8 updates for {op-system-base} workers; those hosts must be replaced with a clean operating system install.
```
Comment (skopacz1) [inline modules/before-updating-web-console.adoc:13]: I'd move this to prereqs for update modules (and ensure parallel wording with other prereqs)

File: modules/before-updating-web-console.adoc [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+* Ensure all Operators previously installed through Operator Lifecycle Manager (OLM) are updated to their latest version in their latest channel. Updating the Operators ensures they have a valid update path when the default OperatorHub catalogs switch from the current minor version to the next during a cluster update.
```
Comment (skopacz1) [inline modules/before-updating-web-console.adoc:14]: I'd move this to prereqs for update modules (and ensure parallel wording with other prereqs)

File: modules/before-updating-web-console.adoc [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+* Ensure that all machine config pools (MCPs) are running and not paused. Nodes associated with a paused MCP are skipped during the update process. You can pause the MCPs if you are performing a canary rollout update strategy.
```
Comment (skopacz1) [inline modules/before-updating-web-console.adoc:15]: I'd move this to prereqs for update modules and reword as follows:

"Ensure that all machine config pools (MCPs) are running and not paused, unless you are performing a canary rollout update strategy. Nodes associated with a paused MCP are skipped during the update process."

File: modules/before-updating-web-console.adoc [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+//remove this???^ or maybe just add another bullet that you can break up the update?
+* To accommodate the time it takes to update, you are able to do a partial update by updating the worker or custom pool nodes. You can pause and resume within the progress bar of each pool.
```
Comment (skopacz1) [inline modules/before-updating-web-console.adoc:17]:
```suggestion
* To accommodate the time it takes to update, you can perform a partial update by updating the worker or custom pool nodes. You can pause and resume within the progress bar of each pool.
```

File: modules/before-updating-web-console.adoc [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+* If your cluster uses manually maintained credentials, update the cloud provider resources for the new release.
```
Comment (skopacz1) [inline modules/before-updating-web-console.adoc:18]: I'd move this to prereqs for update modules (and ensure parallel wording with other prereqs)

File: modules/before-updating-web-console.adoc [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+* Review the list of APIs that were removed in Kubernetes 1.26, migrate any affected components to use the new API version, and provide the administrator acknowledgment.
```
Comment (skopacz1) [inline modules/before-updating-web-console.adoc:19]: I'd move this to prereqs for update modules (and ensure parallel wording with other prereqs)

I think this one is tricky because it might not be in each version's prereqs, and the K8s version is probably different in each OCP version this appears in.

File: modules/before-updating-web-console.adoc [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+
+
+//may need to move to a troubleshooting section
+//* If you run an Operator or you have configured any application with the pod disruption budget, you might experience an interruption during the update process. If `minAvailable` is set to 1 in `PodDisruptionBudget`, the nodes are drained to apply pending machine configs which might block the eviction process. If several nodes are rebooted, all the pods might run on only one node, and the `PodDisruptionBudget` field can prevent the node drain.
```
Comment (skopacz1) [inline modules/before-updating-web-console.adoc:23]: If you do end up moving those other prereqs into relevant modules and this module only has a list of "considerations to make", this would be a good item to uncomment and include in the list above

**Review by skopacz1 — COMMENTED** [review]
Sorry for the long list of comments, but here are my placement and wording suggestions to clean up this assembly-level prereq list. Feel free to disregard if you have other ideas in mind.

File: modules/before-updating-web-console.adoc (hunk 2 above)
Comment (opayne1) [reply modules/before-updating-web-console.adoc:12]: Updated to your suggestion, but wondering if we should specify what "recent enough" means. Leaving this comment thread open to discuss with SMEs.

File: modules/before-updating-web-console.adoc (hunk 8 above)
Comment (opayne1) [reply modules/before-updating-web-console.adoc:19]: I think there is always a list of APIs that are removed each release?? I think we could generalize it and keep it under the considerations module as "check the APIs that were removed" maybe

**Comment by opayne1** [comment]
Hi @wking! Could you take a look at this PR when you have a chance? I am doing this one before I start the PRs to backport the change. Thank you!

File: modules/update-conditional-web-console.adoc [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+.Prerequisites
+* You have access to the cluster with `cluster-admin` privileges.
+
+* You have access to the OpenShift Container Platform web console.
+
+* Your Operators previously installed through Operator Lifecycle Manager (OLM) are updated to their latest version in their latest channel. Updating the Operators ensures they have a valid update path when the default OperatorHub catalogs switch from the current minor version to the next during a cluster update.
+
+* Your machine config pools (MCPs) are running and not paused. Nodes associated with a paused MCP are skipped during the update process. You can pause the MCPs if you are performing a canary rollout update strategy.
+
+* Your {op-system-base}7 workers are replaced with {op-system-base}8 or {op-system} workers before updating to {product-title} {product-version}. Red Hat does not support in-place {op-system-base}7 to {op-system-base}8 updates for {op-system-base} workers; those hosts must be replaced with a clean operating system install.
```
Comment (LalatenduMohanty) [inline modules/update-conditional-web-console.adoc:20]: These content looks disconnected from what we should be communicating. @opayne1 did anyone suggested you the content. Else we can help with the rewrite of the content. cc @wking

**Review by LalatenduMohanty — CHANGES_REQUESTED** [review]

File: modules/update-conditional-web-console.adoc (hunk 10 above)
Comment (opayne1) [reply modules/update-conditional-web-console.adoc:20]: Thanks for taking a look @LalatenduMohanty! This content actually already existed, I just moved it around. We can always move it around or remove it if it is not relevant to this doc.

File: modules/before-updating-web-console.adoc [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+
+* Your etcd backup is recent enough in case your update fails and you must restore your cluster to a previous state.
+
+* You might experience an interruption during the update process if you run an Operator or you have configured any application with the pod disruption budget. If `minAvailable` is set to 1 in `PodDisruptionBudget`, the nodes are drained to apply pending machine configs which might block the eviction process. If several nodes are rebooted, all the pods might run on only one node, and the `PodDisruptionBudget` field can prevent the node drain.
+
+* You might need to update the cloud provider resources for the new release if your cluster uses manually maintained credentials.
+
+* You should provide the administrator acknowledgment and review the list of APIs that were removed with the latest Kubernetes release, and migrate any affected components to use the new API version.
+
+*  You can perform a partial update by updating the worker or custom pool nodes to accommodate the time it takes to update. You can pause and resume within the progress bar of each pool.
```
Comment (wking) [inline modules/before-updating-web-console.adoc:19]: nit: seems like you have an extra space here in `*  You`

File: modules/before-updating-web-console.adoc [hunk 12]
```diff
@@ -0,0 +1,19 @@
...
+
+Before updating using the web console, consider the following:
+
+* Your etcd backup is recent enough in case your update fails and you must restore your cluster to a previous state.
+
+* You might experience an interruption during the update process if you run an Operator or you have configured any application with the pod disruption budget. If `minAvailable` is set to 1 in `PodDisruptionBudget`, the nodes are drained to apply pending machine configs which might block the eviction process. If several nodes are rebooted, all the pods might run on only one node, and the `PodDisruptionBudget` field can prevent the node drain.
+
+* You might need to update the cloud provider resources for the new release if your cluster uses manually maintained credentials.
+
+* You should provide the administrator acknowledgment and review the list of APIs that were removed with the latest Kubernetes release, and migrate any affected components to use the new API version.
```
Comment (wking) [inline modules/before-updating-web-console.adoc:17]: The outgoing:

> * Review the list of APIs that were removed in Kubernetes 1.27, migrate any affected components to use the new API version, and provide the administrator acknowledgment. For more information, see xref:../../updating/preparing_for_updates/updating-cluster-prepare.adoc#updating-cluster-prepare[Preparing to update to {product-title} 4.14].

was possibly more version-specific than we'd like, but I prefer it's opening that focuses on reviewing ack requests instead of providing ack approvals.  We want to reduce the chances that folks blindly click through the acks without actually performing the assessments that the acks are requesting.  Maybe:

* You should review administrator acknowledgement requests, take any recommended actions, and provide the acknowledgement when you are ready.

or something, to keep the opening focus on figuring out what you're being asked to ack?

Admin acks are also a subset of `Upgradeable=False` conditions, so I'm personally fine not discussing admin acks at all here, and instead asking the admin to address any `Upgradeable=False` conditions that the UI might be presenting.

File: modules/update-conditional-web-console.adoc [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+You can view and assess the risks associated with particular updates with conditional updates.
+
+.Prerequisites
+* You have access to the cluster with `cluster-admin` privileges.
+
+* You have access to the OpenShift Container Platform web console.
+
+* Your Operators previously installed through Operator Lifecycle Manager (OLM) are updated to their latest version in their latest channel. Updating the Operators ensures they have a valid update path when the default OperatorHub catalogs switch from the current minor version to the next during a cluster update.
+
+* Your machine config pools (MCPs) are running and not paused. Nodes associated with a paused MCP are skipped during the update process. You can pause the MCPs if you are performing a canary rollout update strategy.
```
Comment (wking) [inline modules/update-conditional-web-console.adoc:18]: or if you're performing an EUS-to-EUS update.  Or if you're performing a control-plane update.  Might be worth describing the full vs. control-plane update UI?

File: modules/before-updating-web-console.adoc [hunk 14]
```diff
@@ -0,0 +1,19 @@
...
+//
+// * updating/updating_a_cluster/updating-cluster-web-console.adoc
+
+:_content-type: CONCEPT
+[id="before-updating-web-console{context}"]
+= Before updating using the web console
+
+Before updating using the web console, consider the following:
+
+* Your etcd backup is recent enough in case your update fails and you must restore your cluster to a previous state.
```
Comment (LalatenduMohanty) [inline modules/before-updating-web-console.adoc:11]: I am not comfortable with suggesting customers restore the cluster from previous etc backup when update status is failing = true. Because failing = true is the indication that update is failing

File: modules/before-updating-web-console.adoc [hunk 15]
```diff
@@ -0,0 +1,19 @@
...
+
+* You might experience an interruption during the update process if you run an Operator or you have configured any application with the pod disruption budget. If `minAvailable` is set to 1 in `PodDisruptionBudget`, the nodes are drained to apply pending machine configs which might block the eviction process. If several nodes are rebooted, all the pods might run on only one node, and the `PodDisruptionBudget` field can prevent the node drain.
```
Comment (LalatenduMohanty) [inline modules/before-updating-web-console.adoc:13]: Also I do not agree with the line `You might experience an interruption during the update process if you run an Operator or you have configured any application with the pod disruption budget` . I am fine with removing this line altogether.

**Review by LalatenduMohanty — CHANGES_REQUESTED** [review]

File: updating/updating_a_cluster/updating-cluster-web-console.adoc [hunk 16]
```diff
@@ -19,49 +19,39 @@ You can update, or upgrade, an {product-title} cluster by using the web console.
...
-* Review the list of APIs that were removed in Kubernetes 1.27, migrate any affected components to use the new API version, and provide the administrator acknowledgment. For more information, see xref:../../updating/preparing_for_updates/updating-cluster-prepare.adoc#updating-cluster-prepare[Preparing to update to {product-title} 4.14].
-* If you run an Operator or you have configured any application with the pod disruption budget, you might experience an interruption during the update process. If `minAvailable` is set to 1 in `PodDisruptionBudget`, the nodes are drained to apply pending machine configs which might block the eviction process. If several nodes are rebooted, all the pods might run on only one node, and the `PodDisruptionBudget` field can prevent the node drain.
+include::modules/before-updating-web-console.adoc[leveloffset=+1]
```
Comment (wking) [inline updating/updating_a_cluster/updating-cluster-web-console.adoc:22]: nit: these prereqs are largely orthogonal to the interface used to request the update.  Even if we don't have the command-line and other process sections include this new prereq module yet, do we want to pick a less-opinionated name like `modules/updating_a_cluster/prerequisites.adoc` and try to keep the prerequisites UI-agnostic?  Also, this is a bit sticky because starting with 4.14, we have [this whole section of pre-update things][1].

image

so if we spend particularly much time talking about prereqs in the process assemblies, we may want to push more of that discussion into prereq assemblies.

[1]: https://docs.openshift.com/container-platform/4.14/updating/preparing_for_updates/updating-cluster-prepare.html

File: updating/updating_a_cluster/updating-cluster-web-console.adoc [hunk 17]
```diff
@@ -19,49 +19,39 @@ You can update, or upgrade, an {product-title} cluster by using the web console.
...
-include::modules/updating-sno.adoc[leveloffset=+1]
+include::modules/update-upgrading-web.adoc[leveloffset=+1]
+
+include::modules/update-conditional-web-console.adoc[leveloffset=+1]
 
 [role="_additional-resources"]
 .Additional resources
 
-* For information on which machine configuration changes require a reboot, see the note in xref:../../architecture/control-plane.adoc#about-machine-config-operator_control-plane[About the Machine Config Operator].
+* xref:../../updating/updating_a_cluster/updating-cluster-cli.adoc#update-conditional-upgrade-pathupdating-cluster-cli[Updating along a conditional update path]
```
Comment (wking) [inline updating/updating_a_cluster/updating-cluster-web-console.adoc:46]: I don't understand why we link to the CLI here.  Can we link like:

```asciidoc
* xref:../../updating/understanding_updates/understanding-update-channels-release.adoc#conditional-updates-overview_understanding-update-channels-releases[Update recommendations and Conditional Updates]
```

or some such to link to the understanding section on conditional updates?

File: modules/before-updating-web-console.adoc (hunk 14 above)
Comment (skopacz1) [reply modules/before-updating-web-console.adoc:11]: Do you want this list item to be removed?

File: modules/before-updating-web-console.adoc (hunk 15 above)
Comment (skopacz1) [reply modules/before-updating-web-console.adoc:13]: Trevor just created [OSDOCS-7661](https://issues.redhat.com/browse/OSDOCS-7661) to drop this item from this list entirely, would you be willing to wait until I start that ticket to make any changes to this item?

File: updating/updating_a_cluster/updating-cluster-web-console.adoc (hunk 16 above)
Comment (skopacz1) [reply updating/updating_a_cluster/updating-cluster-web-console.adoc:22]: I'm a fan of renaming this module to something that's not specific to any update method, like `before-updating-ocp.adoc`. I feel like putting prerequisite in the name might make it confusing to differentiate between these softer considerations and the actual prereqs of the procedures.

Once that "Preparing to update" page gets filled with content other than API acks (or if new assemblies are created), I'm more than happy to help migrate some of the content out of this `before-updating-web-console.adoc` module and into that assembly. But if possible, I'd like to keep that effort separate from this current PR.

**Comment by kalexand-rh** [comment]
The `branch/enterprise-4.15` label has been added to this PR.

This is because your PR targets the `main` branch and is labeled for enterprise-4.14. And any PR going into `main` must also target the latest version branch (enterprise-4.15).

If the update in your PR does NOT apply to version 4.15 onward, please re-target this PR to go directly into the appropriate version branch or branches (enterprise-4.x) instead of main

File: modules/before-updating-web-console.adoc (hunk 15 above)
Comment (opayne1) [reply modules/before-updating-web-console.adoc:13]: Resolving this so I dont accidentally capture this.

File: modules/before-updating-web-console.adoc (hunk 14 above)
Comment (opayne1) [reply modules/before-updating-web-console.adoc:11]: Let me know what you think. I have updated to "You have recently backed up etcd" we can completely remove though if you would prefer.

File: modules/before-updating-web-console.adoc (hunk 12 above)
Comment (opayne1) [reply modules/before-updating-web-console.adoc:17]: Updated! Thanks for the suggestion.

File: modules/update-conditional-web-console.adoc (hunk 13 above)
Comment (opayne1) [reply modules/update-conditional-web-console.adoc:18]: Did a combination of Seb's comments. Lmk what you think.

File: modules/update-conditional-web-console.adoc (hunk 10 above)
Comment (opayne1) [reply modules/update-conditional-web-console.adoc:20]: Removed from this module. Thanks!

File: modules/before-updating-ocp.adoc [hunk 18]
```diff
@@ -0,0 +1,19 @@
+// Module included in the following assemblies:
+//
+// * updating/updating_a_cluster/updating-cluster-web-console.adoc
+
+:_content-type: CONCEPT
+[id="before-updating-ocp{context}"]
+= Before updating the {product-title}
```
Comment (skopacz1) [inline modules/before-updating-ocp.adoc:7]: This phrasing is probably correct in terms of grammar and technical accuracy, but I figure adding cluster to the end would sound better/more in line with our conventions of referring to cluster updates:
```suggestion
= Before updating the {product-title} cluster
```

File: modules/update-conditional-web-console.adoc [hunk 19]
```diff
@@ -0,0 +1,33 @@
...
+
+* Your Operators previously installed through Operator Lifecycle Manager (OLM) are updated to their latest version in their latest channel. Updating the Operators ensures they have a valid update path when the default OperatorHub catalogs switch from the current minor version to the next during a cluster update.
+
+* Your machine config pools (MCPs) are running and not paused. Nodes associated with a paused MCP are skipped during the update process. You can pause the MCPs if you are performing an advanced update strategy, such as a canary rollout, an EUS update, or a control-plane update.
+
+.Procedure
+
+. From the web console, click *Administration* -> *Cluster settings* page and review the contents of the *Details* tab.
+
+. Expand the <need to figure out what UI element is called> to display the potential risks to updating.
```
Comment (opayne1) [inline modules/update-conditional-web-console.adoc:26]: Maybe notification?

Comment (LalatenduMohanty) [reply modules/update-conditional-web-console.adoc:26]: Suggested text: expand the `select new version` UI element

Comment (LalatenduMohanty) [reply modules/update-conditional-web-console.adoc:26]: I am fine with any-better wording as well.

File: modules/update-conditional-web-console.adoc [hunk 20]
```diff
@@ -0,0 +1,33 @@
...
+
+. You can enable `Include supported but not recommended versions` in the `Select new version` dropdown of the *Update cluster* modal to populate the dropdown list with conditional updates.
++
+[NOTE]
+====
+If a `Supported but not recommended` version is selected, an alert will appear below the dropdown menu displaying potential issues with the version.
```
Comment (LalatenduMohanty) [inline modules/update-conditional-web-console.adoc:32]: Substitute `an alert will appear below` with `more information will appear below`

**Review by LalatenduMohanty — CHANGES_REQUESTED** [review]

**Comment by opayne1** [comment]
@LalatenduMohanty I added a couple steps based on the powerpoint you shared with me. Can you let me know what you think?

File: modules/update-conditional-web-console.adoc [hunk 21]
```diff
@@ -0,0 +1,35 @@
...
+. You can enable `Include supported but not recommended versions` in the `Select new version` dropdown of the *Update cluster* modal to populate the dropdown list with conditional updates.
++
+[NOTE]
+====
+If a `Supported but not recommended` version is selected, more information is provided with potential issues with the version.
+====
+
+. Expand the notification to display the potential risks to updating.
+
+. Accept the risk and start the upgrade.
```
Comment (LalatenduMohanty) [inline modules/update-conditional-web-console.adoc:35]: We do not expect customers to accept the risk always. Also we do not recommend them to accept the risk in conditional update.  If they want to update to a version which has some risk they need to access the risk  which is already documented as part of https://docs.openshift.com/container-platform/4.14/updating/preparing_for_updates/updating-cluster-prepare.html#update-preparing-conditional_updating-cluster-prepare

**Review by LalatenduMohanty — CHANGES_REQUESTED** [review]

File: modules/update-conditional-web-console.adoc (hunk 21 above)
Comment (opayne1) [reply modules/update-conditional-web-console.adoc]: Removed this step and edited the previous one to account for this.

File: modules/before-updating-ocp.adoc [hunk 22]
```diff
@@ -0,0 +1,19 @@
...
+
+:_mod-docs-content-type: CONCEPT
+[id="before-updating-ocp{context}"]
+= Before updating the {product-title} cluster
+
+Before updating, consider the following:
+
+* You have recently backed up etcd.
+
+* You might experience an interruption during the update process if you run an Operator or you have configured any application with the pod disruption budget. If `minAvailable` is set to 1 in `PodDisruptionBudget`, the nodes are drained to apply pending machine configs which might block the eviction process. If several nodes are rebooted, all the pods might run on only one node, and the `PodDisruptionBudget` field can prevent the node drain.
```
Comment (LalatenduMohanty) [inline modules/before-updating-ocp.adoc:13]: We must remove the text `You might experience an interruption during the update process if you run an Operator or you have configured any application with the pod disruption budget` as this is not correct.

**Review by LalatenduMohanty — CHANGES_REQUESTED** [review]

**Comment by LalatenduMohanty** [comment]
Apart from https://github.com/openshift/openshift-docs/pull/63315#discussion_r1385214483 , everything looks good to me.

File: modules/before-updating-ocp.adoc (hunk 22 above)
Comment (opayne1) [reply modules/before-updating-ocp.adoc]: Updated!

File: modules/before-updating-web-console.adoc (hunk 15 above)
Comment (opayne1) [reply modules/before-updating-web-console.adoc]: @skopacz1, @LalatenduMohanty and I discussed this. I have removed the sentence requested in this PR, but feel free to edit in however you see fit when you fix the ticket you mentioned.

**Comment by opayne1** [comment]
@shellyyang1989 Could you help with a QE review of this PR? Thank you!

File: updating/updating_a_cluster/updating-cluster-web-console.adoc [hunk 23]
```diff
@@ -19,49 +19,39 @@ You can update, or upgrade, an {product-title} cluster by using the web console.
...
-include::modules/update-changing-update-server-web.adoc[leveloffset=+1]
+If you want to use the canary rollout update process, see xref:../../updating/updating_a_cluster/update-using-custom-machine-config-pools.adoc#update-using-custom-machine-config-pools[Performing a canary rollout update].
+
+include::modules/updating-sno.adoc[leveloffset=+1]
 
 [role="_additional-resources"]
 .Additional resources
 
-* xref:../../updating/understanding_updates/understanding-update-channels-release.adoc#understanding-update-channels-releases[Understanding update channels and releases]
+* For information on which machine configuration changes require a reboot, see the note in xref:../../architecture/control-plane.adoc#about-machine-config-operator_control-plane[About the Machine Config Operator].
```
Comment (abhatt-rh) [inline updating/updating_a_cluster/updating-cluster-web-console.adoc:57]: Per [Additional resources sections](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#additional-resources-sections):

>  You must not include text outside of the square brackets.

```suggestion
* xref:../../architecture/control-plane.adoc#about-machine-config-operator_control-plane[About the Machine Config Operator].
```

File: modules/update-conditional-web-console.adoc [hunk 24]
```diff
@@ -0,0 +1,33 @@
...
+
+* Your Operators previously installed through Operator Lifecycle Manager (OLM) are updated to their latest version in their latest channel. Updating the Operators ensures they have a valid update path when the default OperatorHub catalogs switch from the current minor version to the next during a cluster update.
+
+* Your machine config pools (MCPs) are running and not paused. Nodes associated with a paused MCP are skipped during the update process. You can pause the MCPs if you are performing an advanced update strategy, such as a canary rollout, an EUS update, or a control-plane update.
+
+.Procedure
+
+. From the web console, click *Administration* -> *Cluster settings* page and review the contents of the *Details* tab.
+
+. You can enable `Include supported but not recommended versions` in the `Select new version` dropdown of the *Update cluster* modal to populate the dropdown list with conditional updates.
```
Comment (abhatt-rh) [inline modules/update-conditional-web-console.adoc:26]: I was not able to see these options or dropdown menu on my cluster. Not sure if there was an error setting it up, but I thought I'd mention it for your verification.

**Review by abhatt-rh — COMMENTED** [review]
Nice work, @opayne1 
I added a couple of comments for your consideration. 

/remove-label peer-review-in-progress
/remove-label peer-review-needed
/label peer-review-done

**Comment by shellyyang1989** [comment]
@yanpzhan Could you please help review from UI side? Thank you.

File: modules/update-conditional-web-console.adoc (hunk 24 above)
Comment (opayne1) [reply modules/update-conditional-web-console.adoc:26]: Only impacts the minor (4.y to 4.(y+1))update so depends on what version your cluster was.

**Comment by XiyunZhao** [comment]
The changes look good for me
/label qe-approved

**Comment by yanpzhan** [comment]
Thanks @XiyunZhao for reviewing the pr, the update looks good from UI side.

**Comment by skopacz1** [comment]
LGTM! Thanks again for the help with update docs!
